### ISSUE TITLE
Add support for training with conv2DTranspose

### ISSUE THREAD

**Issue by oeway** [tensorflow/tfjs#449]
Currently we can't run training the Deconvolution layer because the gradients for conv2DTranspose is still missing. 

Here is the error:
```
Uncaught (in promise) Error: Cannot compute gradient: gradient function not found for conv2dDerInput.
```

I posted this issue also here: https://github.com/tensorflow/tfjs/issues/53 but that one is closed.

An alternative to Deconvolution is to use Upsampling+Conv2D, but the gradient for resizeNearestNeighbor is still missing (see https://github.com/tensorflow/tfjs/issues/402 ). 

I really hope one of these solutions could work for training, thanks in advance.

**Comment by tafsiri** [comment]
@oeway not related to conv2DTranspose, but a PR for #402 has just been submitted, so should hopefully not be too long before that hits master.
